packages/block-editor/src/components/resize-canvas/index.js

```diff
@@ -1,20 +1,21 @@
 /**
  * WordPress dependencies
  */
-import { useSimulatedMediaQuery } from '@wordpress/block-editor';
-import { useSelect } from '@wordpress/data';
 import { useEffect, useState } from '@wordpress/element';
 
+/**
+ * Internal dependencies
+ */
+import { default as useSimulatedMediaQuery } from '../../components/use-simulated-media-query';
+
 /**
  * Function to resize the editor window.
  *
+ * @param {string} deviceType Used for determining the size of the container (e.g. Desktop, Tablet, Mobile)
+ *
  * @return {Object} Inline styles to be added to resizable container.
  */
-export function useResizeCanvas() {
-	const deviceType = useSelect( ( select ) => {
-		return select( 'core/edit-post' ).__experimentalGetPreviewDeviceType();
-	}, [] );
-
+export default function useResizeCanvas( deviceType ) {
```

Comment:
Should this be  `use-resize-canvas` in a "hooks" folder instead of "components"?

Reply:
I'm not sure but I can look into it more. I placed it in components since that's where it was in `edit-post`.

Reply:
yeah weird for me, we do have "hooks" folder in some packages.

Reply:
I tried to rework it here as suggested 9b90de0fb5bfd32908fc847493a9b83c438b0b4e, not sure if it's the right way to go about it though. :)